llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -20357,6 +20357,13 @@ ARMTargetLowering::getSingleConstraintMatchWeight(
   return weight;
 }
 
+static bool isIncompatibleReg(const MCPhysReg &PR, MVT VT) {
+  if (VT == MVT::Other)
+    return false;
+  return (ARM::SPRRegClass.contains(PR) && VT != MVT::f32) ||
```

Comment:
```suggestion
  return (ARM::SPRRegClass.contains(PR) && VT != MVT::f32 && VT != MVT::i32) ||
```

We currently accept i32/f32 mismatches, and do the appropriate bitcast.  I think it makes sense to continue doing that.

The other combinations have never worked, as far as I can tell, so probably fine to reject.